### PR TITLE
Remove GUI & GTK

### PR DESCRIPTION
**Pull request by m-horky**
Remove subscription-manager-gui, rhsm-icon and rhsm-gtk packages.

- rhsm-icons are not shared between Cockpit and GUI anymore, maybe they should be brought under the Cockpit package in the future.

### CODE REVIEW

**Comment by jirihnidek** [comment]
I would check if rhsm-icons isn't silently used/required by some other package in RHEL.

File: src/subscription_manager/ga_loader.py [hunk 1]
```diff
@@ -46,46 +43,21 @@
 
 
 class GaImporter(object):
```
Comment (jirihnidek) [inline src/subscription_manager/ga_loader.py:45]: Why don't you remove this completely?

File: Makefile [hunk 2]
```diff
@@ -104,17 +93,9 @@ STYLEFILES=$(PYFILES) $(BIN_FILES)
...
 .DEFAULT_GOAL := build
 
-# Install doesn't perform a build if it doesn't have too.  Best to clean out
-# any cruft so developers don't end up install old builds.
-ifeq ($(WITH_SUBMAN_GUI),true)
-    build: rhsmcertd rhsm-icon
-        EXCLUDE_PACKAGES:="$(EXCLUDE_PACKAGES)" $(PYTHON) ./setup.py clean --all
-        EXCLUDE_PACKAGES:="$(EXCLUDE_PACKAGES)" $(PYTHON) ./setup.py build --quiet --gtk-version=$(GTK_VERSION) --rpm-version=$(VERSION)
-else
-    build: rhsmcertd
```
Comment (jirihnidek) [inline Makefile:114]: I'm sure that we still want to build rhsmcertd!

File: src/subscription_manager/ga_loader.py (hunk 1 above)
Comment (m-horky) [reply src/subscription_manager/ga_loader.py:45]: I didn't do it because GLib is used for dbus communication, and I didn't want to make unneccesary changes in the dbus `server.py` file. I can confirm that importing `gi.repository` objects instead of calling the import method does work as expected, it will be included in the next update of this PR.

The import method was also called in `rhsmcertd_worker.py` and `subscription_manager.py` scripts, but it does not seem it was used by anything.

Comment (jirihnidek) [reply src/subscription_manager/ga_loader.py:45]: Oh, I can see that it is used in e.g. `rhsmcertd_worker.py`, but is this file really necessary to have?

File: src/subscription_manager/ga_impls/ga_gtk3.py [hunk 3]
```diff
@@ -1,22 +1,5 @@
...
-GTK_COMPAT_VERSION = "3"
-
-
-# gtk3 requires constructing with .new(), where
-# gtk2 does not have a .new()
-def tree_row_reference(model, path):
-    return Gtk.TreeRowReference.new(model, path)
-
-
 threads_init = GObject.threads_init
```
Comment (jirihnidek) [inline src/subscription_manager/ga_impls/ga_gtk3.py:5]: I'm almost sure that this file can be removed.

**Comment by m-horky** [comment]
> I would check if rhsm-icons isn't silently used/required by some other package in RHEL.

If [pkgs.org](https://centos.pkgs.org/8/centos-baseos-x86_64/rhsm-icons-1.28.13-3.el8_4.noarch.rpm.html) are to be believed, it is only required by rhsm-gtk and subscription-manager-cockpit.

File: Makefile (hunk 2 above)
Comment (m-horky) [reply Makefile:114]: Are you reading this change correctly? The rhsmcertd is kept there, it was just lifted from the if-else block and the `--gtk-version` argument has been removed.

Comment (jirihnidek) [reply Makefile:110]: Oh, I can see it now. You are right.

File: src/subscription_manager/ga_loader.py (hunk 1 above)
Comment (m-horky) [reply src/subscription_manager/ga_loader.py]: It is not, whole ga_loader logic is removed now.

**Comment by m-horky** [comment]
Before I remove the Draft tag from this PR (and update commit descriptions), i have couple of questions:
- The second commit alters `Requires` statement for SUSE:
https://github.com/candlepin/subscription-manager/blob/36055f48cf18fed0061c6c0fa68596a56e6a00ac/subscription-manager.spec#L285-L288
into
https://github.com/candlepin/subscription-manager/blob/94a30732b8249ae4365158409bfb6f65df57b2dd/subscription-manager.spec#L266-L269
Is this correct? I'm not too familiar with spec syntax
- There are files in `contrib/debian/` which are using some GTK packages. Since they are maintained by community contributors, should they be left alone so they can take care of it themselves?

**Comment by ptoscano** [comment]
Regarding the spec syntax:
```spec
Requires: %{?gtk3:python-gobject} %{!?gtk3:python-gobject2, libzypp, zypp-plugin-python, python-zypp} 
```
is equivalent to write:
```spec
%if %{?gtk3}
Requires: python-gobject
%endif
%if !%{?gtk3}
Requires: python-gobject2, libzypp, zypp-plugin-python, python-zypp
%endif
```
TBH, it feels strange to me that the zypp packages are added only on the non-gtk3 case... but then, it seems the SUSE bits are on python2, which makes them already broken. So I'd put the non-gtk3 case bits just in case, because it feels safer.

**Comment by m-horky** [comment]
The SUSE step was updated to properly list the required packages.

**Comment by jirihnidek** [comment]
@m-horky I cannot remember that suse or debian have ever build any gui variant of susbscription-manager. If there is any requirements on gtk on suse or debian, then it this requirement wasn't 100% correct.

File: setup.py [hunk 4]
```diff
@@ -455,7 +420,6 @@ def find_py(self):
         },
     },
     classifiers=[
-        "Programming Language :: Python :: 2",
```
Comment (ptoscano) [inline setup.py:458]: While technically this is correct, I'd leave it for a different patch series; sadly there's lots more of Python 2 stuff around...

**Comment by ptoscano** [comment]
Except a small note, it looks good to me. Great job :+1:

There are more files that (IMHO) seem to be unused and potential candidate for removals. E.g. `async_utils` (and tests), and `test/rhsm_display.py`. However, since this PR is already big on its own, I'd do these removals in further PR(s), possibly digging it a bit for even more unused stuff.

File: setup.py (hunk 4 above)
Comment (jirihnidek) [reply setup.py:458]: I think that this section `classifiers` doesn't matter, because we don't ship `subscription-manager` on https://pypi.org

Comment (m-horky) [reply setup.py:458]: Yes, I got carried away a bit. Reverting.

**Comment by m-horky** [comment]
`async_utils` is only referenced in tests for it, I think we could remove it now with the rest of GUI. But I agree that it could get cleaned up much more, by removing all the Python 2 bits, in the future PRs.

**Review by jirihnidek — APPROVED** [review]
This PR has my ACK, but please address Pino's requests first.

Thanks for this big PR :+1:

**Review by ptoscano — APPROVED** [review]
Awesome, thanks!!!
